pkg/utils/math.go

```diff
@@ -126,6 +127,32 @@ func FeltDivRem(a, b *fp.Element) (div fp.Element, rem fp.Element) {
 	return div, rem
 }
 
+func Int16FromFelt(n *fp.Element) (int16, bool) {
+	bigN := n.BigInt(new(big.Int))
+	return Int16FromBigInt(bigN)
+}
+
+func Int16FromBigInt(n *big.Int) (int16, bool) {
+	mod := fp.Modulus()
+	negN := new(big.Int).Sub(mod, n)
```

Comment:
Can you please explain this implementation. I don't quite understand why you have to take fp_mod - N. suppose that N is bigger than NegN, does that not imply that N is bigger than maxInt16 ?

Reply:
`n` is representing a felt value, so they don't actually have negative numbers, but we know that if `n > P/2 `then we could use the negative value which is `n - P` (`P` in this case is `mod`). As we want to take the int16 value, I'm just trying to obtain the smaller number possible, so you check which one is smaller, n or negN, and then you compare that one to the highest possible value of int16 to check if the number fits in that type. Does it make sense?

Reply:
Ah yes that makes sense now. Thanks.